Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -93,5 +95,146 @@ public static class CustomEntitySearch
+                KMPTable[0] = -1;
+
+                while (pos < checkMatch.Length)
+                {
+                    if (wordCharArray[pos] == wordCharArray[cmd])
+                    {
+                        KMPTable[pos] = KMPTable[cmd];
+                    }
+                    else
+                    {
+                        KMPTable[pos] = cmd;
+                        cmd = KMPTable[cmd];
+                        while (cmd >= 0 && wordCharArray[pos] != wordCharArray[cmd])
+                        {
+                            cmd = KMPTable[cmd];
+                        }
+                    }
+                    pos++;
+                    cmd++;
+                }
+                KMPTable[pos] = cmd;
+
+                // Begin searching!
+                int currTextChar = 0;
+                int currWordChar = 0;
+                int offset = 0;
+                StringWriter wordFound = new StringWriter();
+                double currMismatch = 0;
+                IList<char> textCharArray = text.ToCharArray();
+
+                while (currTextChar < textCharArray.Count)
+                {
+                    if(wordCharArray[currWordChar] == textCharArray[currTextChar])
+                    {
+                        wordFound.Write(textCharArray[currTextChar]);
+                        currTextChar++;
+                        currWordChar++;
+
+                        if (currWordChar >= wordCharArray.Count)
+                        {
+                            if (currMismatch < leniency && offset > 0)
+                            {
+                                entities.Add(
+                                    new Entity
+                                    {
+                                        Category = "customEntity",
+                                        Value = wordFound.ToString(),
+                                        Offset = currTextChar - (currWordChar + offset),
+                                        Confidence = (currMismatch / leniency)
+                                    });
+                                addWord = true;
+                                currWordChar = KMPTable[currWordChar];
+                                wordFound.Flush();
+                                currMismatch = 0;
+                                offset = 0;
+                            }
+                            else
+                            {
+                                currWordChar = wordCharArray.Count - 1;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        // fuzzy situation?
+                        if (currMismatch < leniency)
+                        {
+                            string accents = @"[\p{Mn}\p{Mc}]";
+                            currMismatch += Regex.IsMatch(wordCharArray[currWordChar].ToString(), accents) ? 0.5 : 1;
+                            offset++;
```

Comment:
check out [GetUnicodeCategory](https://docs.microsoft.com/en-us/dotnet/api/system.char.getunicodecategory?view=netframework-4.8) for an easier-to-read implementation of this

Reply:
And to address whether this is actually doing what it should be doing, take a look at this SO: https://stackoverflow.com/questions/359827/ignoring-accented-letters-in-string-comparison